### DISCUSSION TITLE
스터디 및 과제 순서 변경

### DISCUSSION THREAD

**Discussion by rbgus2002** [rbgus2002/cactusdy#205] [Features]
스터디와 과제에 대해서 순서 변경할 수 있도록 커스텀할 수 있는 기능이 필요하다고 느낍니다 ㅎㅎ
아래에 대해서 이견이 있으시면 말씀 고고 해주세요

### 1. 순서를 변경할 수 있게 할 요소
저는 스터디와 과제 두 개면 된다고 생각하는데, 혹시 놓친게 있을까 해서 여쭤봐요

### 2. 순서 변경 방법
순서 변경이 가능한 요소에 대해서 사용자가 일정 시간 이상 꾹 누르는, 드래그앤드롭 방식이 좋을 거 같아요
참고로 과제 순서를 변경할 때는 개인과제를 그룹과제로 변경을 하는 등의 종류 변경은 불가능하도록 막는게 좋을 거 같습니다. (기획적으로 생각할 게 많아지는 느낌이라)

**Comment by Arkady1356** [comment]
> 저는 스터디와 과제 두 개면 된다고 생각하는데, 혹시 놓친게 있을까 해서 여쭤봐요

네 !! 굳이 추가한다면 Rule도 추가할 수 있겠네요 ㅎㅎ

> 순서 변경이 가능한 요소에 대해서 사용자가 일정 시간 이상 꾹 누르는, 드래그앤드롭 방식이 좋을 거 같아요

저도 좋습니다 Long Press 기반으로 한번 해볼게요..!

> 참고로 과제 순서를 변경할 때는 개인과제를 그룹과제로 변경을 하는 등의 종류 변경은 불가능하도록 막는게 좋을 거 같습니다. 

굿굿입니다. 전에 기획할때도 안그래도 대화했던 부분이라 기억나네요 ㅎㅎ

---

다만 구현 난이도가 조금은 있는 편이라 (좀 많이... 많이 뜯고 분석하고 고쳐야함...)

우선 순위를 어느정도로 하는게 좋을지도 이슈 전반적으로 다루고 결정하면 좋을 것 같아요

**Reply by rbgus2002** [reply]
흠 그렇게 공수가 큰 작업인가요? 
저는 개인적으로 현재 나온 이슈 중에서 제일 먼저 하고싶은 부분이긴 해요

**Reply by Arkady1356** [reply]
네네 어려운 과정입니다
list view라는 일반적인 형태에
애니메이션을 추가하는 것
reloactable하게 만는 것
각각은 Flutter에서 어느정도 지원해줘서 쉽지만, 둘을 동시에 구현하기 위해서는 양쪽 코드 내부를 뜯어보고 합치는 과정을 거쳐야해서...
